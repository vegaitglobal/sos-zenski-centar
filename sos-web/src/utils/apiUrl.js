export const baseUrl = process.env.REACT_APP_BASE_URL;
//CREATE .ENV IN SOS-WEB AND DEFINE REACT_APP_BASE_URL ('http://localhost:48201') 
