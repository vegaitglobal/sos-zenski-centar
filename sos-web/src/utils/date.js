const formatDate = (date) => {
  let validDate = date.toLocaleDateString('en-GB');
  const valid = (() => {
    const day = validDate.split('/')[0];
    const month = validDate.split('/')[1];
    const year = validDate.split('/')[2];
    return `${year}-${month}-${day}`;
  })();
  return valid;
};

export const getLastMonth = () => {
  const today = new Date();

  const firstDay = new Date(today.getFullYear(), today.getMonth() - 1, 1);
  const lastDay = new Date(today.getFullYear(), today.getMonth(), 0);

  return {
    firstDay: formatDate(firstDay),
    lastDay: formatDate(lastDay),
  };
};
