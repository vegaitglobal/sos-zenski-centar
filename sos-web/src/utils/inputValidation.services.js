export const checkEmail = value => {
    return String(value)
        .toLowerCase()
        .match(
            /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
        );
};

export const checkPassword = value => {
    return String(value)
        .match(
            /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])[0-9a-zA-Z!@#$%^&*]{8,}$/
        );
};