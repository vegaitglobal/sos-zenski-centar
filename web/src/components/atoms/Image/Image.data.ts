import { ComponentPropsWithoutRef } from 'react';

export type AspectRatio = [number, number];
export type FitType = 'cover' | 'contain';

type ImageSource = {
	media: string;
	srcset: string;
};

export interface ImageProps extends ComponentPropsWithoutRef<'img'> {
	alt: string;
	src: string;
	fit?: FitType;
	source?: ReadonlyArray<ImageSource>;
	aspectRatio?: AspectRatio;
}

export interface StyledImageProps {
	$aspectRatio?: AspectRatio;
	$hasFit: boolean;
}

export const FALLBACK_SOURCE: ReadonlyArray<ImageSource> = [];
