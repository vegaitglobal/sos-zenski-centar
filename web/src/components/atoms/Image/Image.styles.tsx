import { position, size } from 'polished';
import styled from 'styled-components';
import { aspectRatio } from '../../../styles/helpers/aspectRatio';
import type { FitType, StyledImageProps } from './Image.data';

export const StyledImage = styled.div<StyledImageProps>`
	${size('auto', '100%')};

	${({ $hasFit }) => $hasFit && position('absolute', 0)};
	${({ $aspectRatio }) => $aspectRatio && aspectRatio($aspectRatio)};
`;

export const StyledPicture = styled.picture<{ $fullView: boolean }>`
	${({ $fullView }) => $fullView && position('absolute', 0)};
`;

export const StyledImageElement = styled.img<{ $fit?: FitType }>`
	width: 100%;

	${({ $fit }) => $fit && `height: 100%; object-fit: ${$fit};`}
`;
