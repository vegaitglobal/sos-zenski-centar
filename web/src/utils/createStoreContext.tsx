import { createContext, PropsWithChildren, FunctionComponent, Context, useContext } from 'react';
import useStoreFactory from '../hooks/useStoreFactory';

/**
 * Creates a context for the store created by the passed `storeFactory`.
 * Additionally returns the `Provider` component and the hook for the created store's context.
 *
 * @param name - name of the store
 * @param storeFactory - a function that returns an object representing the store
 */
// eslint-disable-next-line @typescript-eslint/naming-convention
const createStoreContext = <TStore extends Record<string, unknown>>(
	name: string,
	storeFactory: () => TStore,
): [FunctionComponent<unknown>, () => TStore, Context<TStore>] => {
	const context = createContext<TStore>({} as TStore); // the initial value is only to satisfy TS and won't be used

	/**
	 * Initializes and contains the context for the `TStore`.
	 */
	function StoreProvider({ children }: PropsWithChildren<unknown>): JSX.Element {
		const store = useStoreFactory(storeFactory);
		return <context.Provider value={store}>{children}</context.Provider>;
	}

	const useStore = (): TStore => {
		const store = useContext(context);
		if (store) {
			return store;
		}
		throw new ReferenceError(
			`${name} is not initialized, make sure that your component is wrapped in the 'Provider' of the store.`,
		);
	};

	return [StoreProvider, useStore, context];
};

export default createStoreContext;
