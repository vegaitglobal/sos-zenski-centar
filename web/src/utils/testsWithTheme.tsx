import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { ReactElement, ComponentType, ReactNode } from 'react';
import { theme } from '../styles/config/theme';

export const componentWithTheme = (component: ReactNode): ReactElement => (
	<ThemeProvider theme={theme}>{component}</ThemeProvider>
);

export const renderWithTheme = (component: ReactElement, options?: RenderOptions): RenderResult => {
	const wrapper: ComponentType = ({ children }) => componentWithTheme(children);

	return render(component, { wrapper, ...options });
};
