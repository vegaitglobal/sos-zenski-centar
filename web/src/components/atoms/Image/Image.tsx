import { ReactElement } from 'react';
import { FALLBACK_SOURCE, ImageProps } from './Image.data';
import { StyledImage, StyledImageElement, StyledPicture } from './Image.styles';

export const Image = ({
	alt,
	src,
	aspectRatio,
	fit,
	source = FALLBACK_SOURCE,
	...props
}: ImageProps): ReactElement => (
	<StyledImage data-testid="Image" $aspectRatio={aspectRatio} $hasFit={!!fit} {...props}>
		<StyledPicture $fullView={!!aspectRatio || !!fit}>
			{source.map(({ media, srcset }) => (
				<source key={srcset} media={media} srcSet={srcset} />
			))}
			<StyledImageElement data-testid="ImageElement" src={src} alt={alt} $fit={fit} />
		</StyledPicture>
	</StyledImage>
);
