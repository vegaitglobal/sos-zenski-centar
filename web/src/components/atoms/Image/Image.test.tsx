import renderer from 'react-test-renderer';
import Image from './index';
import { renderWithTheme } from '../../../utils/testsWithTheme';

const imageProps = {
	alt: 'background',
	src: 'https://picsum.photos/200',
};

describe('Image', () => {
	describe('Render', () => {
		it('Renders Image', () => {
			const { getByTestId } = renderWithTheme(<Image {...imageProps} />);

			const element = getByTestId('Image');

			expect(element).toBeTruthy();
		});

		it('Renders correctly', () => {
			const tree = renderer.create(<Image {...imageProps} />).toJSON();

			expect(tree).toMatchSnapshot();
		});
	});

	describe('Attributes', () => {
		it('has correct alt text', () => {
			const { getByTestId } = renderWithTheme(<Image {...imageProps} />);

			const element = getByTestId('ImageElement');

			expect(element).toHaveAttribute('alt', imageProps.alt);
		});

		it('has correct source', () => {
			const { getByTestId } = renderWithTheme(<Image {...imageProps} />);

			const element = getByTestId('ImageElement');

			expect(element).toHaveAttribute('src', imageProps.src);
		});
	});
});
