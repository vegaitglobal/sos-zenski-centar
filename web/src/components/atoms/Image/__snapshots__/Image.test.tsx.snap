// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Render Renders correctly 1`] = `
<div
  className="sc-bdnxRM kRerkd"
  data-testid="Image"
>
  <picture
    className="sc-gtsrHT bDSGSX"
  >
    <img
      alt="background"
      className="sc-dlnjwi hGrGGC"
      data-testid="ImageElement"
      src="https://picsum.photos/200"
    />
  </picture>
</div>
`;
